Lib/test/test_imaplib.py

```diff
@@ -458,15 +467,15 @@ def test_simple_with_statement(self):
         with self.imap_class(*server.server_address):
             pass
 
-    @requires_resource('walltime')
+    @expect_timeout()
     def test_imaplib_timeout_test(self):
         _, server = self._setup(SimpleIMAPHandler)
         addr = server.server_address[1]
         client = self.imap_class("localhost", addr, timeout=None)
         self.assertEqual(client.sock.timeout, None)
         client.shutdown()
-        client = self.imap_class("localhost", addr, timeout=support.LOOPBACK_TIMEOUT)
-        self.assertEqual(client.sock.timeout, support.LOOPBACK_TIMEOUT)
+        client = self.imap_class("localhost", addr, timeout=FAST_TIMEOUT)
+        self.assertEqual(client.sock.timeout, FAST_TIMEOUT)
```

Comment:
Should not an explicit timeout be different from `SimpleIMAPHandler.timeout`? Otherwise we cannot know how the value of `client.sock.timeout` was set.

You perhaps can use `expect_timeout()` as a context manager instead of a decorator.

Reply:
I misunderstood the problem with this test -- these connections should not time out. The problem was that the call `self._setup(SimpleIMAPHandler)` created a client and the remainder of the test did not run until that client timedout.

We should use `self._setup(SimpleIMAPHandler, connect=False)`